### DISCUSSION TITLE
Is this legal

### DISCUSSION THREAD

**Discussion by med2404** [xtekky/gpt4free#2142] [General]
Hey everyone 
I get little confusion about this project and i have one simple question before useing it.is it legal ? And thank you all for that.

**Comment by PythonJHH** [comment]
Of course it's legal, it just collects a lot of third-party free interfaces

**Reply by med2404** [reply]
Thank you for your reply.

Can you explain to me how it works? Because I think that you search for free third-party services and then use reverse engineering to get the API they use without any permission.

Is it like that? If not, please explain to me, because I want to use it in my project study, but I am afraid.

**Reply by PythonJHH** [reply]
Your concern is right, but as long as it doesn't cause any impact, I think it's okay
